lib/statsd/instrument/aggregator.rb

```diff
@@ -0,0 +1,261 @@
+# frozen_string_literal: true
+
+module StatsD
+  module Instrument
+    class AggregationKey
+      attr_reader :name, :tags, :no_prefix, :type
+
+      def initialize(name, tags, no_prefix, type)
+        @name = name
+        @tags = tags
+        @no_prefix = no_prefix
+        @type = type
+      end
+
+      def hash
+        [@name, @tags, @no_prefix, @type].hash
+      end
```

Comment:
As @casperisfine pointed out elsewhere, this is another spot for optimization (if it is needed) - we could precompute `hash` in the `initialize` method and make this just an `attr_reader`.

```ruby
attr_reader :name, :tags, :no_prefix, :hash

def initialize(name, tags, no_prefix, type)
  @name = name
  ...
  @hash = [@name, @tags, @no_prefix, @type].hash
end
```

If we see `hash` showing up in profiles, it may be worth it. But if we want to wait and see how it performs first, I think that's okay too.

Reply:
I did this on a branch where i was doing benchmarking and forgot to pull it here. Commited this now.